tests/test_context_window_feature.py

```
@@ -0,0 +1,1132 @@
+            token_count=5000,
+            created_at=now,
+            expires_at=expires_at
+        )
+
+        # Mock the create_checkpoint method directly to avoid serialization complexity
+        with patch('app.services.checkpoint_service.get_supabase', return_value=mock_supabase), \
+             patch('app.services.checkpoint_service.get_redis', return_value=mock_redis):
+            from app.services.checkpoint_service import CheckpointService
+
+            # Create a real service but mock its internal method
+            checkpoint_service = CheckpointService()
```

Ruff (0.14.14)

[error] Local variable `checkpoint_service` is assigned to but never used

Remove assignment to unused variable `checkpoint_service`

(F841)